sqlglot/parser.py

```diff
@@ -3356,7 +3356,7 @@ def _parse_function(
         upper = this.upper()
 
         parser = self.NO_PAREN_FUNCTION_PARSERS.get(upper)
-        if optional_parens and parser:
+        if optional_parens and parser and token_type != TokenType.IDENTIFIER:
```

Comment:
what if it's a string?

Reply:
The simple cases work because `_parse_primary` is invoked before `_parse_function` in `_parse_field`:

```python
➜  sqlglot git:(jo/function_parser_fixup) python
Python 3.10.9 (main, Dec 15 2022, 10:44:50) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sqlglot
>>> sqlglot
>>> sqlglot.parse_one("select 'case'")
(SELECT expressions:
  (LITERAL this: case, is_string: True))
```

But I think you're raising a valid point - I'll come up with a way to make this more robust.